libcasr/src/js.rs

```diff
@@ -0,0 +1,281 @@
+            "at eval (eval at g (/fuzz/FuzzTarget.js:7:7), <anonymous>:8:13)",
+            "at eval (/.svelte-kit/runtime/components/layout.svelte:8:41)",
+        ];
+        let trace = raw_stacktrace
+            .iter()
+            .map(|e| e.to_string())
+            .collect::<Vec<String>>();
+        let sttr = JSStacktrace::parse_stacktrace(&trace);
+        if sttr.is_err() {
+            panic!("{}", sttr.err().unwrap());
+        }
+
+        let stacktrace = sttr.unwrap();
+        assert_eq!(
+            stacktrace[0].debug.file,
+            "/fuzz/node_modules/jpeg-js/lib/decoder.js".to_string()
+        );
+        assert_eq!(stacktrace[0].debug.line, 1110);
+        assert_eq!(stacktrace[0].debug.column, 13);
+        assert_eq!(stacktrace[0].function, "Object.decode".to_string());
+        assert_eq!(stacktrace[1].debug.file, "/fuzz/FuzzTarget.js".to_string());
+        assert_eq!(stacktrace[1].debug.line, 6);
+        assert_eq!(stacktrace[1].debug.column, 14);
+        assert_eq!(stacktrace[1].function, "fuzz".to_string());
+        assert_eq!(
+            stacktrace[2].debug.file,
+            "/fuzz/node_modules/@jazzer.js/core/core.ts".to_string()
+        );
+        assert_eq!(stacktrace[2].debug.line, 335);
+        assert_eq!(stacktrace[2].debug.column, 15);
+        assert_eq!(stacktrace[2].function, "result".to_string());
+        assert_eq!(
+            stacktrace[3].debug.file,
+            "/home/pa_darochek/Documents/SandBox/test_js_stacktrace/main.js".to_string()
+        );
+        assert_eq!(stacktrace[3].debug.line, 1);
+        assert_eq!(stacktrace[3].debug.column, 2017);
+        assert_eq!(stacktrace[3].function, "Worker.fuzz [as fn]".to_string());
+        assert_eq!(stacktrace[4].debug.file, "/home/pa_darochek/.nvm/versions/node/v16.15.1/lib/node_modules/jsfuzz/build/src/worker.js".to_string());
+        assert_eq!(stacktrace[4].debug.line, 55);
+        assert_eq!(stacktrace[4].debug.column, 30);
+        assert_eq!(stacktrace[4].function, "process.<anonymous>".to_string());
+        assert_eq!(stacktrace[5].debug.file, "node:events".to_string());
+        assert_eq!(stacktrace[5].debug.line, 527);
+        assert_eq!(stacktrace[5].debug.column, 28);
+        assert_eq!(stacktrace[5].function, "process.emit".to_string());
+        assert_eq!(stacktrace[6].debug.file, "".to_string());
+        assert_eq!(stacktrace[6].debug.line, 1);
+        assert_eq!(stacktrace[6].debug.column, 2);
+        assert_eq!(stacktrace[6].function, "func".to_string());
+        assert_eq!(stacktrace[7].debug.file, "bootstrap_node.js".to_string());
+        assert_eq!(stacktrace[7].debug.line, 609);
+        assert_eq!(stacktrace[7].debug.column, 3);
+        assert_eq!(stacktrace[7].function, "".to_string());
+        assert_eq!(
+            stacktrace[8].debug.file,
+            "file://<path-to-parent-folder>/offset.js".to_string()
```

Comment:
Should we remove prefix `file:://` during parsing?

Reply:
Done